procedures: Add docs about Open VSX on-premises (#2947)
codenvy/che-docs: modules/administration-guide/pages/running-the-open-vsx-on-premises.adoc

:_content-type: PROCEDURE
:description: Running Open VSX On-Premises
:keywords: administration guide, configuring, openvsx, registry
:navtitle: Running Open VSX On-Premises
:page-aliases:

[id="running-the-open-vsx-on-premises"]
= Running the Open VSX On-Premises

Provides a step-by-step walkthrough for deploying and configuring an on-premises Eclipse Open VSX extension registry, fully integrated with {prod-short} and OpenShift environments. It covers two setup paths - using a {prod-short} workspace or the OpenShift CLI (oc) - to help administrators set up a secure, internal Open VSX instance. This includes creating necessary OpenShift project, deploying Open VSX components, publishing extensions and integrating the registry with {prod-short}.

== Using {prod-short} instance

.Prerequisites

* Be logged in to a {prod-short} as an administrator.
* Internet connection.

.Procedure

. Start a workspace using the Open VSX repository.
+

> Create a workspace using the following link:https://github.com/svor/openvsx/tree/sv-update-os-deployment[Eclipse Open VSX repository].

Author: This link will be updated once the instructions have been reviewed and validated.

----
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "wget -O /tmp/extension.vsix EXTENSION_DOWNLOAD_URL "
----
+
.. Create an extension namespace:
+
[bash,subs="verbatim",options="nowrap"]
---- 
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "ovsx create-namespace EXTENSION_NAMESPACE_NAME" || true
----
+
.. Publish the extension:
+
[bash,subs="verbatim",options="nowrap"]
----
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "ovsx publish /tmp/extension.vsix"
----
+
.. Delete the downloaded extension file:
+ 
[bash,subs="verbatim",options="nowrap"]
----
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "rm /tmp/extension.vsix"
----
+
[TIP]
====

Example: Publish the `redhat.vscode-yaml` extension version 1.18.0:
[bash,subs="verbatim",options="nowrap"]
----
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "wget -O /tmp/extension.vsix https://open-vsx.org/api/redhat/vscode-yaml/1.18.0/file/redhat.vscode-yaml-1.18.0.vsix " &&
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "ovsx create-namespace redhat" || true &&
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "ovsx publish /tmp/extension.vsix" &&
oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "rm /tmp/extension.vsix"

> ----

Reviewer: verified on developer sandbox:

```
2025-08-18 11:10:42 (2.69 MB/s) - '/tmp/extension.vsix' saved [1435588/1435588]

openshift (sv-update-os-deployment) $ oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "ovsx create-namespace redhat" || true &&
> oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "ovsx publish /tmp/extension.vsix" &&
> oc exec -n openvsx "${OVSX_POD_NAME}" -- bash -c "rm /tmp/extension.vsix"
 
🚀  Created namespace redhat
```
<img width="1620" height="580" alt="Screenshot 2025-08-18 at 13 11 35" src="https://github.com/user-attachments/assets/a6cfb1dc-8bb6-4731-9fe9-d606798a49b2" />

====
+
. Verify Open VSX extension registry.
+
Check the list of published extensions by navigating to the URL defined in the OPENVSX_ROUTE_URL environment variable.

== Setting Up Internal Access to the Open VSX Service

> In addition to using a public route to reference the Open VSX registry, it's also possible to configure {prod-short} to set internal cluster service routing. This method provides enhanced security by keeping traffic confined within the cluster, avoiding public exposure.

Reviewer 2: ```suggestion
In addition to using a public route to reference the Open VSX registry, you can also configure {prod-short} to set internal cluster service routing. This method provides enhanced security by keeping traffic confined within the cluster, avoiding public exposure.
```

[id="running-the-open-vsx-on-premises"]
= Running the Open VSX On-Premises

> Provides a step-by-step instructions for deploying and configuring an on-premises Eclipse Open VSX extension registry, fully integrated with {prod-short} and OpenShift environments. It covers two setup paths - using a {prod-short} workspace or the OpenShift CLI (`oc`) - to help administrators set up a secure, internal Open VSX instance. This includes creating necessary OpenShift project, deploying Open VSX components, publishing extensions and integrating the registry with {prod-short}.

Reviewer 2: ```suggestion
Follow the instructions below to deploy and configure an on-premises Eclipse Open VSX extension registry, fully integrated with {prod-short} and OpenShift environments. Choose one of the two setup paths: using a {prod-short} workspace or the OpenShift CLI (`oc`), to help you set up a secure, internal Open VSX instance. This includes creating necessary OpenShift project, deploying Open VSX components, publishing extensions and integrating the registry with {prod-short}.
```
